web3/utils/datastructures.py

```diff
@@ -95,6 +99,15 @@ def __eq__(self, other):
         else:
             return False
 
+    @staticmethod
+    def serializeHexBytes(value):
+        return value.hex() if isinstance(value, HexBytes) else value
+
+    @property
+    def serialized(self):
```

Comment:
@carver curious to hear your opinion on this.  My inclination is to *not* add a new propery/method to the `AttributeDict` class, but to write this as a stand-alone function that people can use, maybe even as something available off of the base `Web3` object.  

No good names immediately come to mind

- `Web3.prepareForJSON`
- `Web3.normalizeForJSON`
- `Web3.makeJSONEncodable`

Thoughts?

Reply:
Since it seems awkward to describe the preparation, and the only thing it's good for is when you're about to encode to json, how about a `Web.toJSON(val)` that does (roughly) this:

```py
# in web3/utils/encoding.py
def to_json(obj):
    prepared = recursive_map(_encode_for_json, obj)
    return json.dumps(prepared)

def _encode_for_json(val):
    if isinstance(val, bytes): return val.hex() #or if we need the prefix, to_hex(val)
    elif isinstance(val, int) and val > JS_MAX_INT: return to_hex(val)
    else: return val

# in web3/main.py
class Web3:
    toJSON = to_json
```